waku/node/jsonrpc/relay/handlers.nim

```diff
@@ -77,14 +84,21 @@ proc installRelayApiHandlers*(node: WakuNode, server: RpcServer, cache: MessageC
     if payloadRes.isErr():
       raise newException(ValueError, "invalid payload format: " & payloadRes.error)
 
-    let message = WakuMessage(
+    var message = WakuMessage(
         payload: payloadRes.value,
         # TODO: Fail if the message doesn't have a content topic
         contentTopic: msg.contentTopic.get(DefaultContentTopic),
         version: msg.version.get(0'u32),
         timestamp: msg.timestamp.get(Timestamp(0)),
         ephemeral: msg.ephemeral.get(false)
       )
+    
+    when defined(rln):
+      if not node.wakuRlnRelay.isNil():
+        let success = node.wakuRlnRelay.appendRLNProof(message, 
+                                                      float64(getTime().toUnix()))
+        if not success:
+          raise newException(ValueError, "Failed to append RLN proof to message")
```

Comment:
Will the `appendRLNProof` print out any more specific error log? This seems to be too ambiguous for users to know what is going on:)

Reply:
I agree, we're working on it on the zerokit side where we return an error enum atleast so the error code is known